api_tests/reviews/mixins/filter_mixins.py

```diff
@@ -259,5 +259,10 @@ def test_review_provider_filters(self, app, url, moderator_pair, admin_pair, exp
         assert expected == actual
 
         # filter by permissions requires auth
-        res = get_actual(app, url, expect_errors=True, permissions='set_up_moderation')
-        assert res.status_code == 401
+        from django.db import transaction
+        try:
+            with transaction.atomic():
+                res = get_actual(app, url, expect_errors=True, permissions='set_up_moderation')
+                assert res.status_code == 401
+        except:
+            pass
```

Comment:
This looks like an actual problem that shouldn't be swept under the rug... If there's an error while handling `/v2/preprint_providers/`, `get_renderer_context` will cause a 500 (instead of a friendly 4xx) by trying to make an additional query.

Maybe it could check whether the response has an error code and, if so, skip the `meta`? Or is there a way to directly check if the current transaction/connection has errored?

```
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/wsgiref/handlers.py", line 85, in run
    self.result = application(self.environ, self.start_response)
  File "/usr/local/lib/python2.7/site-packages/django/contrib/staticfiles/handlers.py", line 63, in __call__
    return self.application(environ, start_response)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/wsgi.py", line 157, in __call__
    response = self.get_response(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 124, in get_response
    response = self._middleware_chain(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 43, in inner
    response = response_for_exception(request, exc)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 93, in response_for_exception
    response = handle_uncaught_exception(request, get_resolver(get_urlconf()), sys.exc_info())
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/exception.py", line 41, in inner
    response = get_response(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 249, in _legacy_get_response
    response = self._get_response(request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 187, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/usr/local/lib/python2.7/site-packages/django/core/handlers/base.py", line 185, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/utils/decorators.py", line 185, in inner
    return func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
    return view_func(*args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/django/views/generic/base.py", line 68, in view
    return self.dispatch(request, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 491, in dispatch
    self.response = self.finalize_response(request, response, *args, **kwargs)
  File "/usr/local/lib/python2.7/site-packages/rest_framework/views.py", line 416, in finalize_response
    response.renderer_context = self.get_renderer_context()
  File "/code/api/preprint_providers/views.py", line 85, in get_renderer_context
    'whitelisted_providers': [unicode(provider) for provider in SharePreprintProviderWhitelisted.objects.all()]
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 250, in __iter__
    self._fetch_all()
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 1118, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python2.7/site-packages/django/db/models/query.py", line 53, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch)
  File "/usr/local/lib/python2.7/site-packages/django/db/models/sql/compiler.py", line 894, in execute_sql
    raise original_exception
TransactionManagementError: An error occurred in the current transaction. You can't execute queries until the end of the 'atomic' block.
[09/Jan/2018 14:10:14] "GET /v2/preprint_providers/?filter[permissions]=set_up_moderation HTTP/1.1" 500 59
```